fiboa_cli/convert_utils.py

```diff
@@ -466,3 +132,402 @@ def read_geojson(path, **kwargs):
     obj["features"] = list(map(normalize_geojson_properties, obj["features"]))
 
     return gpd.GeoDataFrame.from_features(obj, crs = "EPSG:4326")
+
+
+class BaseConverter:
+    bbox: Optional[tuple[float]] = None
+    id: str = None
+    short_name: str = None
+    title: str = None
+    license: str | dict[str, str] = None
+    attribution: str = None
+    description: str = None
+    providers: list[dict] = None
+
+    sources: Optional[dict[str, str] | str] = None
+    source_variants: Optional[dict[dict[str, str] | str]] = None
+    variant: str = None
+
+    columns: dict[str, str] = None
+    column_additions: dict[str, str] = EMPTY_DICT
+    column_filters: dict[str, callable] = EMPTY_DICT
+    column_migrations: dict[str, callable] = EMPTY_DICT
+    missing_schemas: dict[str, str] = EMPTY_DICT
+    extensions: list[str] = tuple()
```

Comment:
Shouldn't that better be a set instead of a tuple to avoid duplicates?

Reply:
I did type `list[str]` but then added instance `tuple()` as a kind-of immutable list. `set()` could also work, than I would initialize it with a `frozenset()`.

Reply:
I think we just have very different approaches in mind. I find it weird that I can't add additional things to the set, which I may want to do depending on inputs.

Reply:
Yes, I just fixed the data-sharing issue. Do you want me to make it a set?

Reply:
Ported to #147